app/services/id_token_builder.rb

```diff
@@ -55,24 +57,47 @@ def timestamp_claims
   end
 
   def acr
-    ial = identity.ial
-    case ial
-    when Idp::Constants::IAL_MAX then determine_ial_max_acr
-    when Idp::Constants::IAL1 then Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF
-    when Idp::Constants::IAL2 then Saml::Idp::Constants::IAL2_AUTHN_CONTEXT_CLASSREF
-    else
-      raise "Unknown ial #{ial}"
+    return nil unless identity.acr_values.present?
+    acr_ial_component_values.map do |component_value|
+      if component_value == Vot::LegacyComponentValues::IALMAX
+        determine_ial_max_acr.name
+      else
+        component_value.name
+      end
+    end.join(' ')
+  end
+
+  def sp_requests_vot?
```

Comment:
We probably also want to add a `vot.present?` clause here as well. That way we won't end up with a `vtm` claim when there is not `vot`.